src/engine/Match_rules.ml

```diff
@@ -161,8 +162,9 @@ let per_rule_boilerplate_fn ~timeout ~timeout_threshold =
     | Some res -> res
     | None ->
         incr cnt_timeout;
+        rule_timeouts := rule_id :: !rule_timeouts;
```

Comment:
We have a utility function for this. Previously called `Common.push` but I bet with the recent refactoring it's now `List_.push`